Added vectorized Riemann solvers for the single-layer shallow water equations

chaulio opened the pull request:
These changes introduce new vectorized implementations for the single-layer shallow water equations (chile2010 example). The user can easily switch between the serial and the vectorized code: to use vectorization, one needs to define GEOCLAW_VEC=1 before compiling. More info in the chile2010 Makefile.

mandli commented on src/rpn2_geoclaw_vec.f90:
```diff
@@ -0,0 +1,403 @@
+         hvL=qr(i-1,nv) 
+         hvR=ql(i,nv)
+         
+         sw1 = 0.0d0
```
You may want to add a comment as to why you needed to decompose this into scalars for future reference.

chaulio replied:
Done!

mandli commented on src/geoclaw_riemann_utils_vec.f90:
```diff
@@ -0,0 +1,698 @@
+      integer meqn,mwaves,maxiter
+      !double precision fw(meqn,mwaves)
+      !double precision sw(mwaves)
+      double precision sw1,sw2,sw3,fw11,fw12,fw13,fw21,fw22,fw23,fw31,fw32,fw33
```
We have been using `real(kind=8)` instead of `double precision` with the intent to perhaps support multiple precisions depending on the case.  I know the former is less portable but I think it's probably best to be consistent unless there is another reason to use `double precision`.

chaulio replied:
I've now replaced double precision with real(kind=8) in my code. Note however that the original geoclaw_riemann_utils.f still uses double precision, you may want to check whether this should also be modified.

mandli commented on src/rpn2_geoclaw_vec.f90:
```diff
@@ -0,0 +1,403 @@
+      use amr_module, only: mcapa
+
+      use storm_module, only: pressure_forcing, pressure_index
+      use statistics
```
What is this module?

chaulio replied:
It was a module I created to measure the execution time of each solver, but I removed it before this pull request. Apparently, I forgot to delete it from a couple of files, and my compilation tests did not complain because the .mod file was still present in my computer. I have already removed these "use statistics" lines from the code.

chaulio commented:
I have pushed a new commit that addresses the issues you have pointed out. 

Main changes:
* double precision -> real(kind=8) ;
* !DIR$ SIMD -> !$OMP SIMD ;
* Added a few comments explaining that I had to decompose some arrays into several scalars because of what seems to be a bug with the Intel Compiler;
* Removed subroutines riemann_ssqfwave and riemann_fwave from geoclaw_riemann_utils_vec.f90 because they were not being used and also were not compatible with the vectorized code (different order in the array indices);
* Removed "use statistics", which is a module I was using to measure the execution time taken by each solver.

mandli commented:
This looks good to me now.  Randy can you test this quick and make sure it also works for you?

rjleveque commented:
It compiles and runs, but seems to be slower.  See the comment I posted on clawpack/geoclaw#291.
